### PR TITLE
Add ability for TSI to inspect loadmem flags

### PR DESCRIPTION
**Pull request by colinschmidt**
TSI also marks that memory as preloaded

### CODE REVIEW

File: fesvr/tsi.cc [hunk 1]
```diff
@@ -16,6 +16,12 @@ void tsi_t::host_thread(void *arg)
 tsi_t::tsi_t(int argc, char** argv) : htif_t(argc, argv)
 {
   target = context_t::current();
+  loadmem_addr = 0;
+  for (int i = 1; i < argc; i++) {
+      std::string arg(argv[i]);
+      if (arg.find("+loadmem_addr=") == 0)
```
Comment (zhemao) [inline fesvr/tsi.cc:22]: Does this work correctly if +loadmem_addr is inside +permissive/+permissive-off?

File: fesvr/tsi.cc [hunk 2]
```diff
@@ -113,3 +119,11 @@ void tsi_t::tick(bool out_valid, uint32_t out_bits, bool in_ready)
   if (in_valid() && in_ready)
     in_data.pop_front();
 }
+
+bool tsi_t::is_address_preloaded(addr_t taddr, size_t len)
+{
+  if(loadmem_addr != 0)
```
Comment (zhemao) [inline fesvr/tsi.cc:125]: I think you should keep a separate boolean flag to indicate whether something has been loaded or not.

File: fesvr/tsi.cc (hunk 1 above)
Comment (zhemao) [reply fesvr/tsi.cc:22]: It's also possible that +loadmem_addr is passed but +loadmem isn't. In which case you wouldn't want to return true in is_address_preloaded.

Comment (colinschmidt) [reply fesvr/tsi.cc:22]: The only way you can pass those arguments is between +permissive. That is the way I've been testing it.
I'll add the extra check that loadmem was also passed in.

File: fesvr/tsi.cc (hunk 2 above)
Comment (colinschmidt) [reply fesvr/tsi.cc:125]: Will do.

**Pull request merged**